### ISSUE TITLE
connection to api.github.com/repos/keepassxreboot

### ISSUE THREAD

**Issue by rezad1393** [keepassxreboot/keepassxc-browser#1832]
I see random connections to 
https://api.github.com/repos/keepassxreboot/keepassxc/releases/latest

even though I have set "Check for updates of KeePassXC:" never

**Comment by varjolintu** [comment]
This shouldn't be possible. In the code the automatic check is done only if the option is set to something else than "Never". However, if you press the "Check for updates" button manually, of course the connection is made to the URL. Can you somehow pinpoint when exactly the connections happen? While loading the settings page?

**Comment by rezad1393** [comment]
I saw the connection by chance when I have browser console open and saw it randomly.

I saw that in setting it is set to never, maybe addon updates resets that?

so I flipped to a week and then back to never and will monitor.

**Comment by varjolintu** [comment]
Addon updates shouldn't reset any settings. Only if the setting is not present at all. Hopefully this is some rare situation where localHost has corrupted "slightly" and deleted some settings. If this happens again, I'd recommend creating a new profile.

**Comment by acmondor** [comment]
I observed this problem as well. In my case it was very is easy to reproduce as it happened every time I started firefox. I debugged the issue and it turned to be an issue with the value of checkUpdateKeePassXC. That setting was set to "0" which caused run-time tests to fail because they are expecting the setting to have a numerical value (ie 0 and not "0").

I resolved this by going into the extensions settings and selecting a new value for "Check for updates of KeePassXC:", eg "every month", then changed it back to "never". This changed the setting to 0 from "0" and made the run-time tests behave as expected (ie check for updates won't be run).

If you want to confirm that your current setting is "0" before you actually change anything, go to the extension's settings and click on "Export settings" to save them in a file. Afterwards open view the saved file in a text editor or just dump in a shell with cat and look for the value of checkUpdateKeePassXC. In my case it was the first item and before the fix it was saved as:

`
    "checkUpdateKeePassXC":"0"
`

After the fix it was saved as:

`
    "checkUpdateKeePassXC":0
`

If anyone is interested in observing this run-time behaviour for themselves they can do this (before the above mentioned fix is performed):

1) Edited your hosts file (eg /etc/hosts Linux) and add the following line to force a failure of the update check:
`
        127.0.0.1    api.github.com
`
   * Note: This may not actually be required for what follows, but it is something I did early to stop to the connections to api.github.com and may help with the reproducibility of what follows:

2) Debug the extension as follows:
```

    Launch firefox, open new browser tab, then go to:
        about:debugging#/runtime/this-firefox

    Click on "Inspect" next to "KeePassXC-Browser"

    In the Debugger select:  Main Thread >> KeePassXC-Browser >> background >> keepass.js

    Put breakpoint at the "if" statement in the following code (ie click on 803 linue number):

        802 | keepass.keePassXCUpdateAvailable = function() {
        803 |     if (page.settings.checkUpdateKeePassXC && page.settings.checkUpdateKeePassXC !== CHECK_UPDATE_NEVER) {

    * Note: Your line numbers may be different if you're running a different version of the
            extension. I'm running Version 1.8.4 from December 20, 2022.

    Then opening a new blank tab should be enough to cause the breakpoint to be hit. Once it's
    hit hover the mouse over page.settings.checkUpdateKeePassXC will show that it's value is
    "0" and hovering over CHECK_UPDATE_NEVER will show its value is 0 (without the quotes).

    Then stepping into the code you will see that the if condition does fail and an attempt
    is made to check for an update.

```
3) When you are all done, edited your hosts file and undo the changes you make in step 1.

Last, but not least, I have no idea how checkUpdateKeePassXC got set to "0" in the first place. Perhaps something changed in the extension of firefox itselt that affected the way settings are saved and/or loaded.
